tests/MongoDB.Driver.Core.Tests/Core/Configuration/ClusterBuilderTests.cs

```diff
@@ -59,10 +63,56 @@ public void CreateServerMonitorFactory_should_return_expected_result(int connect
 
             var serverSettings = result._settings();
         }
+
+        [Fact]
+        public void RegisterStreamFactory_should_complement_previous_RegisterStreamFactory_call_in_right_order()
+        {
+            var subject = new ClusterBuilder();
+            subject.RegisterStreamFactory(
+                factory => {
+                    var testFactory = (StreamFactoryTest)factory;
+                    testFactory.TestValue.Should().Be(1);
+                    testFactory.TestValue++;
+                    return factory;
+                });
+            subject.RegisterStreamFactory(
+                factory => {
+                    var testFactory = (StreamFactoryTest)factory;
+                    testFactory.TestValue.Should().Be(2);
+                    testFactory.TestValue++;
+                    return factory;
+                });
+            subject.RegisterStreamFactory(
+                factory => {
+                    var testFactory = (StreamFactoryTest)factory;
+                    testFactory.TestValue.Should().Be(3);
+                    testFactory.TestValue++;
+                    return factory;
+                });
+            var streamFactoryWrapper = subject._streamFactoryWrapper();
+            streamFactoryWrapper(new StreamFactoryTest { TestValue = 1 });
+        }
+    }
+
+    public class StreamFactoryTest : IStreamFactory
```

Comment:
Maybe call this class FakeStreamFactory?

We've used the name FakeXyz before for fake objects that are used for testing.

Reply:
Sounds good,